src/server/service/auth.ts

```diff
@@ -95,7 +94,7 @@ export default class AuthService implements IAuthService {
   }
 
   async register(dto: RegisterDTO): Promise<void> {
-    await this.userService.create(dto);
+    await this.userService.create(dto, 'admin@admin.com');
   }
```

Comment:
This doesn't look quite right, why hardcoding this email here? This should be the data for the "creator" user, not just the email.

Another issue Im spotting here is that the parameters order doesn't look right, check on your definition for `userService.create` here:

https://github.com/Davejs136/thruway-parking/blob/2faa5ac782653b0935a654216d5cef9276739b46/src/server/service/user.ts#L75

You should provide `currentUser` which is an instance of `User` as the first parameter, and then an object of `CreateUserDTO` type as a second parameter.